src/countdown-timer/countdown-timer.tsx

```diff
@@ -0,0 +1,172 @@
+import debounce from "lodash/debounce";
+import { useEffect, useMemo, useRef, useState } from "react";
+import { useMediaQuery } from "react-responsive";
+import { useTimer } from "../util/use-timer";
+import { CountdownTimerProps } from "./types";
+import {
+    Countdown,
+    PositionStyle,
+    Time,
+    TimeLeft,
+    Wrapper,
+} from "./countdown-timer.style";
+import { TimeHelper } from "../util/time-helper";
+import { ClockIcon } from "@lifesg/react-icons";
+import { useIntersectionObserver } from "../util/use-intersection-observer";
+import { useEventListener } from "../util/use-event-listener";
+import { MediaWidths } from "../spec/media-spec";
+
+export const CountdownTimer = ({
+    className,
+    timer: _timer,
+    notifyTimer,
+    offset,
+    show,
+    "data-testid": testId,
+    onDuration,
+    onNotify,
+    onFinish,
+}: CountdownTimerProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const { minutes = 0, seconds = 0 } = _timer;
+    const notifyMinutes = notifyTimer?.minutes || 0;
+    const notifySeconds = notifyTimer?.seconds || 0;
+    const notifyTiming = notifyMinutes * 60 + notifySeconds;
+    const totalSeconds = minutes * 60 + seconds;
+
+    const wrapperRef = useRef<HTMLDivElement>();
+    const stickyRef = useRef<HTMLDivElement>();
+
+    const [scrolling, setScrolling] = useState<boolean>(false);
+    const [offsetY, setOffsetY] = useState<number>(0);
+
+    const [timer, isPlaying, setIsPlaying] = useTimer(totalSeconds);
+    const onScreen = useIntersectionObserver(stickyRef, {
+        threshold: 1,
+        rootMargin: `${offsetY * -1}px 0px 0px 0px`,
+    });
+
+    const isMobile = useMediaQuery({
+        maxWidth: MediaWidths.mobileL,
+    });
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+
+    useEventListener("scroll", handleScrollEvent);
+
+    useEffect(() => {
+        setIsPlaying(show);
+    }, [show]);
+
+    useEffect(() => {
+        if (timer === 0) {
+            performOnFinishHandler();
+        } else if (timer === notifyTiming) {
+            performOnDurationHandler();
+            performOnNotifyHandler();
```

Comment:
there's no guarantee the timer will run on time (e.g. tab is put is background so it gets paused), so I suggest to be safe, `if (timer < notifyTiming)`, call `onNotify` if it has not been called